### PR TITLE
docs(compiler-hooks): adds examples, formatting and missing descriptions

### PR DESCRIPTION
**Pull request by misterdev**
Adds some examples and improves formatting and hooks descriptions.

### CODE REVIEW

**Comment by misterdev** [comment]
Do you mind reviewing this? :) @EugeneHlushko @montogeek

File: src/content/api/compiler-hooks.md [hunk 1]
```diff
@@ -164,83 +193,90 @@ Parameters: `compilation`, `compilationParams`
...
-Parameters: `compilation`
+- Callback Parameters: `compilation`
 
 
 ### `shouldEmit`
 
 `SyncBailHook`
 
-Can return true/false at this point
+Called before emitting assets, should return a boolean to say wheter ot not emit.
```
Comment (EugeneHlushko) [inline src/content/api/compiler-hooks.md]:
```suggestion
Called before emitting assets. Should return a boolean telling whether to emit.
```

File: src/content/api/compiler-hooks.md [hunk 2]
```diff
@@ -49,77 +52,84 @@ For the description of hook types, see [the Tapable docs](https://github.com/web
...
-Parameters: `compiler`
+- Callback Parameters: `compiler`
 
 
 ### `run`
 
 `AsyncSeriesHook`
 
-Hook into the compiler before it begins reading records.
+Hook into the compiler before it begins reading ["records"](https://webpack.js.org/configuration/other-options/#recordspath).
```
Comment (EugeneHlushko) [inline src/content/api/compiler-hooks.md]:
```suggestion
Hook into the compiler before it begins reading [`records`](/configuration/other-options/#recordspath). 
```

File: src/content/api/compiler-hooks.md [hunk 3]
```diff
@@ -49,77 +52,84 @@ For the description of hook types, see [the Tapable docs](https://github.com/web
 
 `SyncBailHook`
 
-Executes a plugin after [the `entry` configuration](https://webpack.js.org/configuration/entry-context/#entry) from webpack options has been processed.
+Called after the [`entry` configuration](https://webpack.js.org/configuration/entry-context/#entry) from webpack options has been processed.
```
Comment (EugeneHlushko) [inline src/content/api/compiler-hooks.md]:
```suggestion
Called after the [`entry` configuration](/configuration/entry-context/#entry) from webpack options has been processed.
```

File: src/content/api/compiler-hooks.md [hunk 4]
```diff
@@ -49,77 +52,84 @@ For the description of hook types, see [the Tapable docs](https://github.com/web
...
+
+- Callback Parameters: [`context`](https://webpack.js.org/configuration/entry-context#context), [`entry`](https://webpack.js.org/configuration/entry-context#entry)
```
Comment (EugeneHlushko) [inline src/content/api/compiler-hooks.md]:
```suggestion
- Callback Parameters: [`context`](/configuration/entry-context/#context), [`entry`](/configuration/entry-context/#entry)
```

**Comment by misterdev** [comment]
Thanks for reviewing @EugeneHlushko, I've commited your suggestions. Let me know if there's anything else to improve!

Also, if you like those stuff I have some more in #2852 😇

**Comment by EugeneHlushko** [comment]
Thanks!

**Comment by EugeneHlushko** [comment]
Please mention me in PR so i get notified at some point

**Comment by misterdev** [comment]
Thanks to you for reviewing!

**Pull request merged**